inst/templates/monolix-step-01-preprocess.Rmd

```diff
@@ -0,0 +1,180 @@
+---
+title: "Monolix Data Processing"
+author: "Your Name"
+date: "Created: YYYY-MM-DD, Last run: `r Sys.Date()`"
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+
+options(knitr.kable.NA = '-')
+
+library(tidyverse)
+library(glue)
+library(here)
+library(knitr)
+library(kableExtra)
+library(usethis)
+```
+
+## Install and check the data package
+
+<!-- check against data specs: https://github.com/FredHutch/VISC-Assay-Methods/tree/master/PK/Assay_Data_Specs -->
+
+```{r install-data-package, eval=FALSE}
+
+remotes::install_git("N:/cavd/studies/cvdxxx/pdata/xxx.git")
+```
+
+```{r load-data-package}
+
+# Load PK concentrations and assign to pk_data 
+library(Ho785)
+data("Ho785_covance_elisa_pk")
+
+pk_data <- Ho785_covance_elisa_pk
+
+glimpse(pk_data)
+```
+
+<!-- The STP should create a table of missing concentrations in the QC report. You may want to further investigate patterns of missingness here. -->
+
+```{r check-for-missing-concentrations}
+
+# this will only work if there is one row per expected visit in the data package
+pk_data %>% 
+  filter(is.na(concentration)) %>% 
+  group_by(pubid) %>% 
+  summarise(
+    missing_visits = paste0(visit_type, collapse = ", ")
+  ) %>% 
+  kable() %>% 
+  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
+```
+
+<!-- Next, check dose and covariate data. -->
+
+```{r check-dose}
+
+# these variables are used to calculate AMT
+pk_data %>% 
+  distinct(group, analyte, route, dose, dose_units) %>% 
+  arrange(group, analyte) %>% 
+  kable() %>% 
+  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
+```
+
+```{r check-units}
+
+# check other units that are used for calculating covariate values
+
+pk_data %>% distinct(weight_unit, height_unit)
+```
+
+<!-- Usually, we evaluate participant weight, BMI, age, sex as covariates in the PK model. Baseline serum albumin is another covariate worth investigating. Bajaj et. al (2019) found that baseline body weight, serum albumin, and sex are the most commonly identified covariates in oncology PK models (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6710515/).  -->
+
+<!-- At pre-infusion time points, concentrations of the analyte should be LLoQ. If not, plot these data and have a discussion about how you will take it into account in the PK model. -->
+
+```{r check-pre-infusion}
+
+# check for concentrations > LLOQ at pre-infusion time points
+pk_data %>% 
+  filter(visit_type == "1.00 A") %>% 
```

Comment:
This may be protocol specific as well. SRA needs to check study schema to verify which `visit_type` corresponds to baseline.